tests/conftest.py

```diff
@@ -392,29 +409,47 @@ def _make_sdata_for_testing_querying_and_aggretation() -> SpatialData:
             "by_circles": by_circles,
         },
     )
-    # to visualize the cases considered in the test, much more immediate than reading them as text as done above
-    PLOT = False
-    if PLOT:
-        ##
-        import matplotlib.pyplot as plt
-
-        ax = plt.gca()
-        sdata.pl.render_shapes(element="values_polygons", na_color=(0.5, 0.2, 0.5, 0.5)).pl.render_points().pl.show(
-            ax=ax
-        )
-        sdata.pl.render_shapes(element="values_circles", na_color=(0.5, 0.2, 0.5, 0.5)).pl.show(ax=ax)
-        sdata.pl.render_shapes(element="by_polygons", na_color=(1.0, 0.7, 0.7, 0.5)).pl.show(ax=ax)
-        sdata.pl.render_shapes(element="by_circles", na_color=(1.0, 0.7, 0.7, 0.5)).pl.show(ax=ax)
-        plt.show()
-        ##
 
     # generate table
-    x = np.ones((21, 2)) * np.array([1, 2])
+    x = RNG.random((21, 1))
     region = np.array(["values_circles"] * 9 + ["values_polygons"] * 12)
     instance_id = np.array(list(range(9)) + list(range(12)))
-    table = AnnData(x, obs=pd.DataFrame({"region": region, "instance_id": instance_id}))
+    categorical_obs = pd.Series(pd.Categorical(["a"] * 9 + ["b"] * 9 + ["c"] * 3))
+    numerical_obs = pd.Series(RNG.random(21))
+    table = AnnData(
+        x,
+        obs=pd.DataFrame(
+            {
+                "region": region,
+                "instance_id": instance_id,
+                "categorical_in_obs": categorical_obs,
+                "numerical_in_obs": numerical_obs,
+            }
+        ),
+        var=pd.DataFrame(index=["numerical_in_var"]),
+    )
     table = TableModel.parse(
         table, region=["values_circles", "values_polygons"], region_key="region", instance_key="instance_id"
     )
     sdata.table = table
+    # to visualize the cases considered in the test, much more immediate than reading them as text as done above
+    if plot:
+        import matplotlib.pyplot as plt
+        import spatialdata_plot
+
+        # otherwise pre-commits will remove the "unused" import
+        _ = spatialdata_plot
+
+        ax = plt.gca()
+        sdata.pl.render_shapes(element="values_polygons", color="categorical_in_obs").pl.show(ax=ax)
+        sdata.pl.render_shapes(element="values_circles", color="categorical_in_obs").pl.show(ax=ax)
+        sdata.pl.render_shapes(element="by_polygons", na_color=(1.0, 0.7, 0.7, 0.3)).pl.show(ax=ax)
+        sdata.pl.render_shapes(element="by_circles", na_color=(1.0, 0.7, 0.7, 0.3)).pl.show(ax=ax)
+        sdata.pl.render_points(color="categorical_in_ddf", size=10.0, palette="tab10").pl.show(ax=ax)
+        plt.show()
     return sdata
```

Comment:
I don't think we want to have data plotting to be a dependency of this, I would remove this

Reply:
agree, we don't to have it as a dependency. It was useful for developing the tests, but I will move it to the sandbox.